### PR TITLE
Bashiir

### PR DESCRIPTION
**Pull request by bashiir**
added unit test

### CODE REVIEW

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java [hunk 1]
```diff
@@ -113,6 +127,48 @@ public void testGetTasks() throws Exception {
...
+        String sDate="2020-03-22";
+        SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd");
+        Date date=dateFormat.parse(sDate);
+        start.setDate(date);
+        Task task1 = createTaskWithDueDate(start,"task1");
+        String body = (new ObjectMapper()).valueToTree(start).toString();
+        mockMvc.perform(MockMvcRequestBuilders.post("/tasks/due/start")
+            .with(user(USERNAME))
+            .contentType(MediaType.APPLICATION_JSON)
+                .content(body)
```
Comment (smthwas) [inline BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: The request body is unnecessary for the API.

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java [hunk 2]
```diff
@@ -53,6 +61,9 @@
 
     @Autowired
     private TaskRepository taskRepository;
+    @Autowired
+    private TaskManagerService taskService;
```
Comment (smthwas) [inline BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: I didn't see the reference of this attribute. Do we need it?

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java [hunk 3]
```diff
@@ -113,6 +127,48 @@ public void testGetTasks() throws Exception {
...
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$", hasSize(1)))
+            .andExpect(jsonPath("$[0].scheduledstart", is("2020-03-23")));
+           sDate="2020-04-22";
+        date=dateFormat.parse(sDate);
+        body = (new ObjectMapper()).valueToTree(start).toString();
+        mockMvc.perform(MockMvcRequestBuilders.post("/tasks/due/start")
+                .with(user(USERNAME))
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(body)
```
Comment (smthwas) [inline BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: Same as the above.

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java [hunk 4]
```diff
@@ -113,6 +127,48 @@ public void testGetTasks() throws Exception {
...
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$", hasSize(0)));
+        mockMvc.perform(MockMvcRequestBuilders.post("/tasks/due/start")
+                .with(user("unknown"))
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(body)
```
Comment (smthwas) [inline BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: Same as the above.

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java (hunk 1 above)
Comment (bashiir) [reply BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: You can't send the request without body. it errors out. try it.

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java (hunk 4 above)
Comment (bashiir) [reply BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: same as the first comment

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java (hunk 3 above)
Comment (bashiir) [reply BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: same as the first comment

Comment (bashiir) [reply BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: same as the first comment

File: BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java (hunk 1 above)
Comment (smthwas) [reply BACKEND/userAuth/src/test/java/com/apptime/auth/controller/TaskManagerTest.java]: Oh sorry it is my mistake. I thought it was a different api. never mind.

**Pull request merged**